sdk/eventhub/eventhubs-checkpoint-blob/src/blobPartitionManager.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { PartitionManager, PartitionOwnership, Checkpoint } from "@azure/event-hubs";
+import { ContainerClient } from "@azure/storage-blob";
+import * as log from "./log";
+import { throwTypeErrorIfParameterMissing } from "./util/error";
+
+/**
+ * An implementation of PartitionManager that uses Azure Blob Storage to persist checkpoint data.
+ * @class
+ */
+export class BlobPartitionManager implements PartitionManager {
+  private _containerClient: ContainerClient;
+
+  constructor(containerClient: ContainerClient) {
+    this._containerClient = containerClient;
+  }
+  /**
+   * Get the list of all existing partition ownership from the underlying data store. Could return empty
+   * results if there are is no existing ownership information.
+   * Partition Ownership contains the information on which EventProcessor is currently processing the partition and the last checkpoint for the partition
+   *
+   * @param eventHubName The event hub name.
+   * @param consumerGroupName The consumer group name.
+   * @return Partition ownership details of all the partitions that have had an owner.
+   */
+  async listOwnership(
+    eventHubName: string,
+    consumerGroupName: string
+  ): Promise<PartitionOwnership[]> {
+    const partitionOwnershipArray: PartitionOwnership[] = [];
+    try {
+      for await (const blob of this._containerClient.listBlobsFlat({
+        include: ["metadata"],
+        prefix: `${eventHubName}/${consumerGroupName}/`
+      })) {
+        const blobPath = blob.name.split("/");
+        const blobName = blobPath[blobPath.length - 1];
+        const partitionOwnership: PartitionOwnership = {
+          eventHubName,
+          consumerGroupName,
+          ownerId: blob.metadata!.ownerid,
```

Comment:
We should make sure to keep an eye out on #4966

Could you create an issue that references the above so we make sure we update this package when we update our storage dependency?

Reply:
Yes sure! Created an issue- https://github.com/Azure/azure-sdk-for-js/issues/5042.